packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2003,17 +2023,41 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         proposalHandle: string | undefined,
         ackHandle: string,
         summaryLogger: ITelemetryLogger,
+        clientId: string,
     ) {
+        // Tracks whether the summarizer node downloads the snapshot corresponding to the ack.
+        let ackedSnapshotDownloaded = false;
+
+        const getSnapshotP = new LazyPromise<ISnapshotTree>(async () => {
+            ackedSnapshotDownloaded = true;
+            return this.fetchSnapshotFromStorage(ackHandle, summaryLogger, {
+                eventName: "RefreshLatestSummaryGetSnapshot",
+                fetchLatest: false,
+            });
+        });
+
         const readAndParseBlob = async <T>(id: string) => readAndParse<T>(this.storage, id);
         await this.summarizerNode.refreshLatestSummary(
             proposalHandle,
-            async () => this.fetchSnapshotFromStorage(ackHandle, summaryLogger, {
-                eventName: "RefreshLatestSummaryGetSnapshot",
-                fetchLatest: false,
-            }),
+            async () => getSnapshotP,
             readAndParseBlob,
             summaryLogger,
         );
+
+        /**
+         * If the acked snapshot was downloaded by the summarizer node and the ack is for a summary that was not
+         * generated by this client, it's possible that it had a different GC version than the current. If so, update
+         * the summary GC version which will update the last successful summary that this client has seen. We will
+         * regenerate the summary and GC data in the next summary to make it up to date to the current version.
+         */
+         if (ackedSnapshotDownloaded && clientId !== this.clientId) {
+            const snapshot = await getSnapshotP;
+            const blobId = snapshot.blobs[metadataBlobName];
+            if (blobId) {
+                const metadata = await readAndParse<IContainerRuntimeMetadata>(this.storage, blobId);
+                this.summaryGCVersion = gcFeature(metadata);
+            }
+        }
```

Comment:
There could be another case where `clientId === this.clientId` and that triggers `this.summaryGCVersion = this.currentGCVersion` or something like that.

Reply:
Fixed it.